Code/Framework/AzToolsFramework/Tests/PaintBrush/PaintBrushPaintLocationTests.cpp

```diff
@@ -0,0 +1,339 @@
+                        AZ::Vector3(dirtyArea.GetCenter().GetX(), dirtyArea.GetMin().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetMin().GetX(), dirtyArea.GetCenter().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetCenter().GetX(), dirtyArea.GetCenter().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetMax().GetX(), dirtyArea.GetCenter().GetY(), 0.0f),
+                        AZ::Vector3(dirtyArea.GetCenter().GetX(), dirtyArea.GetMax().GetY(), 0.0f),
+                    };
+                    EXPECT_THAT(validPoints, ::testing::Pointwise(ContainerIsClose(), expectedValidPoints));
+
+                    // We should only have 5 opacities, and they should all be 1.0 because we haven't adjusted any brush settings.
+                    EXPECT_EQ(opacities.size(), 5);
+                    for (auto& opacity : opacities)
+                    {
+                        EXPECT_NEAR(opacity, 1.0f, 0.001f);
+                    }
+
+                    // Validate the blendFn:
+                    // By default, it should be a normal blend, so verify that a base value of 0, a new value of 1, and an opacity of 0.5
+                    // gives us back a blended value of 0.5.
+                    const float blend = blendFn(0.0f, 1.0f, 0.5f);
+                    EXPECT_NEAR(blend, 0.5f, 0.001f);
+                });
+
+
+        paintBrush.BeginPaintMode();
+        paintBrush.BeginBrushStroke(m_settings);
+        paintBrush.PaintToLocation(TestBrushCenter, m_settings);
+        paintBrush.EndBrushStroke();
+        paintBrush.EndPaintMode();
+    }
+
+    TEST_F(PaintBrushPaintLocationTestFixture, PaintToLocationWithSmallMovementDoesNotTriggerPainting)
+    {
+        // This verifies that if the distance between two PaintToLocation calls is small enough, it won't trigger
+        // an OnPaint. "small" is defined as less than (brush size * distance %).
+
+        AzToolsFramework::PaintBrush paintBrush(EntityComponentIdPair);
+        ::testing::NiceMock<MockPaintBrushNotificationBusHandler> mockHandler(EntityComponentIdPair);
+
+        // Set the distance between brush stamps to 50%
+        const float TestDistancePercent = 50.0f;
+        m_settings.SetDistancePercent(TestDistancePercent);
+
+        // Set the brush radius to 1 meter (diameter is 2 meters)
+        const float TestBrushRadius = 1.0f;
+        m_settings.SetSize(TestBrushRadius * 2.0f);
+
+        // The distance we expect to need to move to trigger another paint call is 50% of our brush size.
+        const float DistanceToTriggerSecondCall = TestBrushRadius * 2.0f * (TestDistancePercent / 100.0f);
+
+        // Choose a second brush center location that's just slightly under the threshold that should be needed to trigger a second
+        // OnPaint call.
+        const AZ::Vector3 tooSmallSecondLocation = TestBrushCenter + AZ::Vector3(DistanceToTriggerSecondCall - 0.01f, 0.0f, 0.0f);
```

Comment:
question: Could these be tied to constants somewhere as these values might change in future potentially? If you think this is unlikely it's fine to leave as-is

Reply:
Can you explain what you mean a bit more here? 

The way the paintbrush works is that it draws circles every N pixels, where N is determined by the radius of the circle and the Distance % setting. This test is verifying that if you've moved _less_ than (BrushSize * Distance%), then a circle won't get drawn, but if you move at least that amount, then it will. 

The only constant at this point in the test is the "- 0.01", which is just an arbitrary small amount. This test should work equally well if that's changed to "- 1" or "- 0.00001". (At some point float precision will come into play, but you get the idea)

I'm not sure what you're suggesting should be turned into constants outside of the test here...

Reply:
Ah okay understood - yes I really just meant the magic number here (`0.01`) that might be an important minimum that could be named and comes up as a tolerance in the main paintbrush code. If it's totally arbitrary then it's not a big deal, I just didn't want to miss the opportunity that something could be moved to a named constant if it made sense 🙂

Reply:
Thx for explaining! Yep, it's a 100% totally arbitrary constant here, it has nothing to do with the main paintbrush code. Guess I should improve the comment here to make that more explicit. :)